components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/CacheBackedApplicationDAO.java

```diff
@@ -160,6 +166,37 @@ public String getServiceProviderNameByClientId(String clientId, String type, Str
         return appName;
     }
 
+    public String getApplicationResourceIDByInboundKey(String inboundKey, String inboundType, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        String resourceId;
+        ApplicationResourceIDCacheInboundAuthKey cacheKey = new ApplicationResourceIDCacheInboundAuthKey(inboundKey,
+                inboundType, tenantDomain);
+        ApplicationResourceIDCacheInboundAuthEntry entry = resourceIDCacheByInboundAuth.getValueFromCache(cacheKey,
+                tenantDomain);
+        if (entry != null) {
+            resourceId = entry.getApplicationResourceId();
+            if (resourceId != null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Resource ID is present in the cache for " + inboundKey);
```

Comment:
Can we introduce a toString method for ApplicationResourceIDCacheInboundAuthKey and use it to log here